src/Mobs/Monster.h

```diff
@@ -274,8 +300,23 @@ class cMonster :
 	/** Adds weapon that is equipped with the chance saved in m_DropChance[...] (this will be greter than 1 if picked up or 0.085 + (0.01 per LootingLevel) if born with) to the drop */
 	void AddRandomWeaponDropItem(cItems & a_Drops, unsigned int a_LootingLevel);
 
+	/** Entity leashed to */
+	cEntity * m_LeashedTo;
+
+	/** Entity pos leashed to, for leash knots */
+	Vector3d * m_LeashToPos;
```

Comment:
Oh, sorry, that's smelling of memory leak. 
I will change it to a regular member. Is there any benefit of using unique_ptr instead of a regular member here? Thanks